packages/flutter/lib/src/material/dialog.dart

```diff
@@ -469,12 +473,23 @@ class AlertDialog extends StatelessWidget {
       }
     }
 
+    // The paddingScaleFactor is used to adjust the edge padding of Dialog
+    // children.
+    final double paddingScaleFactor = _paddingScaleFactor(MediaQuery.of(context).textScaleFactor);
+    final TextDirection textDirection = Directionality.of(context);
+
     Widget titleWidget;
     Widget contentWidget;
     Widget actionsWidget;
-    if (title != null)
-     titleWidget = Padding(
-        padding: titlePadding ?? EdgeInsets.fromLTRB(24.0, 24.0, 24.0, content == null ? 20.0 : 0.0),
+    if (title != null) {
+      final EdgeInsets defaultTitlePadding = titlePadding?.resolve(textDirection) ?? EdgeInsets.fromLTRB(24.0, 24.0, 24.0, content == null ? 20.0 : 0.0);
```

Comment:
It might be a little clearer like this:
```dart
final EdgeInsets defaultTitlePadding = EdgeInsets.fromLTRB(24.0, 24.0, 24.0, content == null ? 20.0 : 0.0);
final EdgeInsets effectiveTitlePadding = titlePadding?.resolve(textDirection) ?? defaultTitlePadding;
```

Reply:
Done